main.py

```diff
@@ -66,77 +77,73 @@ def players_settings():
 
         color = s.textinput(players_names[i]+", choose color!", "Enter letter:\n\nb - black \ng - green \np - pink"
                                                                     " \nr - red \ny - yellow \no - orange\n")
-        while color not in {'b', 'g', 'p', 'r', 'y', 'o'}:
+        while color not in colors_list:
             color = s.textinput(players_names[i]+", choose color!", "Enter correct letter:\n\nb - black \ng - green"
                                                                         " \np - pink \nr - red \ny - yellow "
                                                                         "\no - orange\n")
-            if color in {'b', 'g', 'p', 'r', 'y', 'o'}:
-                    break                
-        num[i].fillcolor(color_dict[color])
-        num[i].clear()
-        num[i].shape("turtle")
-        num[i].shapesize(3, 3, 3)
-        num[i].penup()
-        num[i].goto(-300, x[i])
-        num[i].pendown()
-        s.update()
-        s.tracer(1)
-        
+            if color in colors_list:
+                break
+
+        num[i].fillcolor(color_dict[color.lower()])        
+        turtle_starting_pos(i)
+   
+
+       
 
 def rematch_settings():
     dice_image.res_reset()
-    buttons.match_result.clear()
-    s.tracer(0)
+    buttons.match_result.clear()  
     
     for i in range(number_of_players):
-        
-        num[i].clear()
-        num[i].shape("turtle")
-        num[i].shapesize(3, 3, 3)
-        num[i].penup()
-        num[i].goto(-300, x[i])
-        num[i].pendown()
-        
-    s.update()
-    s.tracer(1)    
+        turtle_starting_pos(i)
+            
 
 def dice_result(_, __):
 
     dice_image.res_reset()
+    buttons.roll_button.hideturtle()
     
     result = str(random.choice([1, 2, 3, 4, 5, 6]))
     dice_dict[result]()
     
     s.listen()
     dice_results.append(result)
 
-    a = len(dice_results) % number_of_players
-    if a == 0:
-        a = number_of_players
 
-    num[a-1].fd(int(result)*step_size)
+    player_turn = len(dice_results) % number_of_players
+    if player_turn == 0:
+        player_turn = number_of_players
+
+    num[player_turn-1].fd(int(result)*step_size)
+
+    buttons.roll_button.showturtle()
+
+    
 
     for i in range(number_of_players):
         if num[i].xcor() >= 250:
             buttons.match_result.write(str(players_names[i])+' wins!', font=("Verdana", 20, "normal"), align='center')
             replay()
+
         
 
 def replay():
     replay = s.textinput("Play again?", "Type 'Y' to play again, type 'N' to exit")
 
     if replay:

```

Comment:
Why the code doesn't check if replay is `Y` or `N` at the beggining, but after some `if/elif` checks? If it was checked at the beggining I think you could delete one `if/elif` check